Sources/Common/Core/DataArray/index.js

```diff
@@ -207,6 +231,24 @@ function vtkDataArray(publicAPI, model) {
     }
   };
 
+  publicAPI.insertTuple = (idx, tuple) => {
+    if (model.size <= idx * model.numberOfComponents) {
+      model.size = (idx + 1) * model.numberOfComponents;
+      reallocate(model.size);
+    }
+    publicAPI.setTuple(idx, tuple);
+  };
+
+  publicAPI.insertNextTuple = (tuple) => {
+    const idx = model.size / model.numberOfComponents;
+    model.size += model.numberOfComponents;
+    if (model.size >= model.values.length) {
+      reallocate(model.values.length * 2);
+    }
```

Comment:
`return publicAPI.insertTuple(model.size / model.numberOfComponents, tuple);` ?

Reply:
Changed it 👍🏼 

However, something that I don't quite understand:

`model.size` is the number of actually filled tuples, correct? (We use `model.size` in `insertNextTuple` to determine where to put the next tuple.)

But once we go over `model.values.length`, resize sets `model.size` to `2*model.values.length`. But then `insertNextTuple` would set the next tuple at the end of the array again, no?

Reply:
I am asking, because in the C++ code they set `this->Size = numTuples * numComps`, but I am guessing in vtk.js we want to keep `model.size` as the currently filled values.

Reply:
> But once we go over model.values.length, resize sets model.size to 2*model.values.length. But then insertNextTuple would set the next tuple at the end of the array again, no?

`resize()` shouldn't modify `model.size`